dhis-2/dhis-api/src/main/java/org/hisp/dhis/deduplication/DeduplicationStatus.java

```diff
@@ -27,10 +27,26 @@
  */
 package org.hisp.dhis.deduplication;
 
+import lombok.Getter;
+
+@Getter
 public enum DeduplicationStatus
 {
-    ALL,
-    OPEN,
-    INVALID,
-    MERGED
+    ALL( true, false, false ),
+    OPEN( true, true, false ),
+    INVALID( true, false, false ),
+    MERGED( true, false, true );
+
+    private final boolean forFilter;
```

Comment:
I would not add these fields for the following reasons:

* `forFilter` is never used. All status have `forFilter` set to true.
* `forCreation` is only true for OPEN. Most of the code added in DefaultDeduplicationService can be replaced by just stating that the status has to be OPEN when creating a potential duplicate.
* `forMerge` is only true for MERGE. So it just repeats what we already know.

so to me, this adds complexity (duplicate information) that is not needed.